api/api.go

```diff
@@ -664,3 +668,853 @@ func (f *PublicController) SearchCourse(c *gin.Context) {
+		if err != nil {
+			log.Errorf("Unable to convert parameter `deregister_deadline` to time.Time: %s", err.Error())
+			c.Status(http.StatusBadRequest)
+			return
+		}
+		deregisterDeadline = deregisterDeadline.Local()
+	}
+
+	pCtrl := exam.PublicController{Database: f.Database}
+	err = pCtrl.EditExam(name, description, date, duration, examId, userId, online, null.StringFrom(location), null.TimeFrom(registerDeadline), null.TimeFrom(deregisterDeadline))
+	if err != nil {
+		log.Errorf("Unable to edit exam: %s", err.Error())
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+	c.Status(http.StatusOK)
+}
+
+func (f *PublicController) UploadExamFile(c *gin.Context) {
+	id, err := strconv.Atoi(c.Param("id"))
+	if err != nil {
+		log.Errorf("Unable to convert parameter `id` to int: %s", err.Error())
+		c.Status(http.StatusBadRequest)
+		return
+	}
+
+	user_id, err := f.GetIdFromCookie(c)
+	if err != nil {
+		log.Errorf("Unable to get id from Cookie: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+
+		return
+	}
+
+	if c.ContentType() == "text/plain" {
+		type _file struct {
+			Name string `json:"name"`
+			Uri  string `json:"uri"`
+		}
+
+		var file _file
+		if err := c.BindJSON(&file); err != nil {
+			log.Errorf("Unable to bind json: %s\n", err.Error())
+			c.IndentedJSON(http.StatusBadRequest, err.Error())
+			return
+		}
+		log.Info(file)
+
+		pCtrl := exam.PublicController{Database: f.Database}
+		err = pCtrl.UploadExamFile(file.Name, file.Uri, user_id, id, false, nil)
+		if err != nil {
+			log.Errorf("Unable to create Exam-URI: %s", err.Error())
+			c.Status(http.StatusInternalServerError)
+			return
+		}
+	} else {
+		file, err := c.FormFile("file")
+		if err != nil {
+			log.Errorf("No file found in request: %s", err.Error())
+			c.Status(http.StatusBadRequest)
+			return
+		}
+
+		fi, err := file.Open()
+		if err != nil {
+			log.Errorf("Unable to open file: %s", err.Error())
+			c.Status(http.StatusInternalServerError)
+			return
+		}
+
+		pCtrl := exam.PublicController{Database: f.Database}
+		err = pCtrl.UploadExamFile(file.Filename, "", user_id, id, true, fi)
+		if err != nil {
+			log.Errorf("Unable to create ExamFile: %s", err.Error())
+			c.Status(http.StatusInternalServerError)
+			return
+		}
+	}
+
+	c.Status(http.StatusCreated)
+}
+
+func (f *PublicController) GetExamById(c *gin.Context) {
+	id, err := strconv.Atoi(c.Param("id"))
+	if err != nil {
+		c.Status(http.StatusInternalServerError)
+		return
+	}
+	// Fetch Data from Database with Backend function
+	pCtrl := exam.PublicController{Database: f.Database}
+	co, err := pCtrl.GetExamByID(id)
+	if err != nil {
+		log.Errorf("Unable to get exam: %s\n", err.Error())
+		c.IndentedJSON(http.StatusBadRequest, err.Error())
+		return
+	}
+	// Return Status and Data in JSON-Format
+	c.IndentedJSON(http.StatusOK, co)
+}
+
+func (f *PublicController) GetRegisteredExamsFromUser(c *gin.Context) {
```

Comment:
Autorisierung fehlt

Reply:
Durch den DB-Zugriff in exam.go wird implizit schon alles geholt, was zur Cookie-id passt. Da ist keine Autorisierung nötig.